DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs

```diff
@@ -2182,4 +1566,196 @@ internal class ContextMenuCommand
+                    .GroupBy(ac => ac.Name).Select(x => x.First()).Where(ac => ac.Name.StartsWith(context.Options[0].Value.ToString(), StringComparison.OrdinalIgnoreCase));
+                var list = slashCommands.ToList();
+                foreach (var sc in list.Take(25))
+                {
+                    options.Add(new DiscordApplicationCommandAutocompleteChoice(sc.Name, sc.Name.Trim()));
+                }
+                return options.AsEnumerable();
+            }
+        }
+
+        public class DefaultHelpAutoCompleteLevelOneProvider : IAutocompleteProvider
+        {
+            public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Provider(AutocompleteContext context)
+            {
+                var options = new List<DiscordApplicationCommandAutocompleteChoice>();
+                var globalCommands = await context.Client.GetGlobalApplicationCommandsAsync();
+                var guildCommands = await context.Client.GetGuildApplicationCommandsAsync(context.Guild.Id);
+                var slashCommands = globalCommands.Concat(guildCommands)
+                .Where(ac => !ac.Name.Equals("help", StringComparison.OrdinalIgnoreCase))
+                .GroupBy(ac => ac.Name).Select(x => x.First());
+                var command = slashCommands.FirstOrDefault(ac =>
+                    ac.Name.Equals(context.Options[0].Value.ToString().Trim(),StringComparison.OrdinalIgnoreCase));
+                if (command is null || command.Options is null)
+                {
+                    options.Add(new DiscordApplicationCommandAutocompleteChoice("no_options_for_this_command", "no_options_for_this_command"));
+                }
+                else
+                {
+                    var opt = command.Options.Where(c => c.Type is ApplicationCommandOptionType.SubCommandGroup or ApplicationCommandOptionType.SubCommand
+                                                         && c.Name.StartsWith(context.Options[1].Value.ToString(), StringComparison.InvariantCultureIgnoreCase)).ToList();
+                    foreach (var option in opt.Take(25))
+                    {
+                        options.Add(new DiscordApplicationCommandAutocompleteChoice(option.Name, option.Name.Trim()));
+                    }
+                }
+                return options.AsEnumerable();
+            }
+        }
+
+        public class DefaultHelpAutoCompleteLevelTwoProvider : IAutocompleteProvider
+        {
+            public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Provider(AutocompleteContext context)
+            {
+                var options = new List<DiscordApplicationCommandAutocompleteChoice>();
+                var globalCommands = await context.Client.GetGlobalApplicationCommandsAsync();
+                var guildCommands = await context.Client.GetGuildApplicationCommandsAsync(context.Guild.Id);
```

Comment:
Same comment regarding firing off at the same time -- depending on rate limiting